### ISSUE TITLE
complete_check() does not return TRUE for mapped input 

### ISSUE THREAD

**Issue by blahgeek** [vim/vim#5547]
**Describe the bug**
`complete_check()` does not return TRUE for mapped input. Aka, mapped input cannot interrupt long running complete function via `complete_check()`. 

**To Reproduce**

```vim
set nocompatible

" Some complete function that may take a long time (taken from |E840|)
fun! CompleteMonths(findstart, base)
    if a:findstart
        " locate the start of the word
        let line = getline('.')
        let start = col('.') - 1
        while start > 0 && line[start - 1] =~ '\a'
            let start -= 1
        endwhile
        return start
    else
        " find months matching with "a:base"
        for m in split("Jan Feb Mar Apr May Jun Jul Aug Sep Oct Nov Dec")
            if m =~ '^' . a:base
                call complete_add(m)
            endif
            sleep 300m	" simulate searching for next match
            if complete_check()
                echom 'complete check 1'
                break
            endif
        endfor
        return []
    endif
endfun

set completefunc=CompleteMonths
set completeopt+=menuone,noselect

inoremap <SPACE> <SPACE><SPACE>
```

```
vim -u init.vim
F<C-x><C-u><SPACE>
```

### Actual behaviour
`complete_check()` does not return TRUE. The complete function takes 3+ seconds to return.

### Expected behaviour
`complete_check()` return TRUE. The complete function takes 300ms to return.

**Environment (please complete the following information):**
 - Vim version 8.2

**Comment by chrisbra** [comment]
Yes, I believe this happens, to be able to replay macros. (Macros and mapped keys are handled similar, which means `complete_check()` only returns true, if the key was actually typed.)

**Comment by chrisbra** [comment]
Note, the documentation talks about a **key typed** (from `:h complete_check()`):

> Check for a **key typed** while looking for completion matches.
> This is to be used when looking for matches takes some time.
> Returns TRUE when searching for matches is to be aborted,
> zero otherwise.
> Only to be used by the function specified with the
> 'completefunc' option.

**Comment by blahgeek** [comment]
OK. But is it possible to detect that it's triggered directly from an actual typed key?
There's many popular plugins that map keys in insert mode (e.g., https://github.com/jiangmiao/auto-pairs will map `SPACE`, `(`, `)` and so on), which make these keys cannot interrupt the complete function.

**Comment by chrisbra** [comment]
I am not sure. Why do you need this?

**Comment by blahgeek** [comment]
As I mentioned above, I use https://github.com/jiangmiao/auto-pairs which would map `<SPACE>` and some other keys in insert mode. In which case, hitting `<SPACE>` while complete function is running will block until the complete function finishes.
